OpenRA.Game/Map/Map.cs

```diff
@@ -285,6 +286,24 @@ public static string ComputeUID(IReadOnlyPackage package)
 			}
 		}
 
+		public static int GetFormat(IReadOnlyPackage p)
+		{
+			Dictionary<string, MiniYaml> yaml;
+			using (var yamlStream = p.GetStream("map.yaml"))
+			{
+				if (yamlStream == null)
+					throw new FileNotFoundException("Required file map.yaml not present in this map");
+
+				yaml = new MiniYaml(null, MiniYaml.FromStream(yamlStream, "map.yaml", stringPool: Game.ModData.MapCache.StringPool)).ToDictionary();
+			}
+
+			yaml.TryGetValue("MapFormat", out var temp);
+			var format = FieldLoader.GetValue<int>("MapFormat", temp.Value);
+			if (format < Map.SupportedMapFormat)
+				throw new InvalidDataException($"Map format {format} is not supported.");
+			return format;
+		}
+
```

Comment:
I'm not sure how to exactly handle it. In a few places I need map format in order to generate uid, but map format is not yet extracted from the map. Here I just copied some code from `MapPreview` and made it a static function